src/entities/pair.ts

```diff
@@ -10,38 +10,45 @@ import { InsufficientReservesError, InsufficientInputAmountError } from '../erro
 export const computePairAddress = ({
   factoryAddress,
   tokenA,
-  tokenB
+  tokenB,
+  initHashCode
 }: {
   factoryAddress: string
   tokenA: Token
   tokenB: Token
+  initHashCode: string
 }): string => {
   const [token0, token1] = tokenA.sortsBefore(tokenB) ? [tokenA, tokenB] : [tokenB, tokenA] // does safety checks
   return getCreate2Address(
     factoryAddress,
     keccak256(['bytes'], [pack(['address', 'address'], [token0.address, token1.address])]),
-    INIT_CODE_HASH
-  )
+    initHashCode
 }
 export class Pair {
   public readonly liquidityToken: Token
   private readonly tokenAmounts: [CurrencyAmount<Token>, CurrencyAmount<Token>]
+  private readonly factoryAddress: string
+  private readonly initCodeHash: string
 
-  public static getAddress(tokenA: Token, tokenB: Token, factoryAddress = FACTORY_ADDRESS): string {
-    return computePairAddress({ factoryAddress, tokenA, tokenB })
+  public static getAddress(tokenA: Token, tokenB: Token, factoryAddress = FACTORY_ADDRESS, initHashCode = INIT_CODE_HASH): string {
+    return computePairAddress({ factoryAddress, tokenA, tokenB, initHashCode })
   }
 
   public constructor(
     currencyAmountA: CurrencyAmount<Token>,
     tokenAmountB: CurrencyAmount<Token>,
-    factoryAddress = FACTORY_ADDRESS
+    factoryAddress = FACTORY_ADDRESS,
+    initCodeHash: string,
```

Comment:
Don't think it's legal to do non-optional argument after optional. Better to make both optional.